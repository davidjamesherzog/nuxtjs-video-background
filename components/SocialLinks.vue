<template>
  <ul class="social" data-social-links>
    <li>
      <a href="#" data-social-links-facebook
        ><img src="~/assets/images/facebook.png" data-social-links-facebook-img
      /></a>
    </li>
    <li>
      <a href="https://twitter.com/@videojs" data-social-links-twitter>
        <img src="~/assets/images/twitter.png" data-social-links-twitter-img />
      </a>
    </li>
    <li>
      <a href="#" data-social-links-instagram
        ><img
          src="~/assets/images/instagram.png"
          data-social-links-instagram-img
      /></a>
    </li>
  </ul>
</template>

<style lang="scss" scoped>
.social {
  position: absolute;
  z-index: 10;
  bottom: 20px;
  display: flex;
  justify-content: center;
  align-items: center;

  li {
    list-style: none;

    a {
      display: inline-block;
      margin-right: 20px;
      filter: invert(1);
      transform: scale(0.5);
      transition: 0.5s;

      &:hover {
        transform: scale(0.5) translateY(-15px);
      }
    }
  }
}
</style>
