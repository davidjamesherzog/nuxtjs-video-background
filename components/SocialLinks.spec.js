import { mount } from '@vue/test-utils';
import SocialLinks from '@/components/SocialLinks.vue';

describe('SocialLinks', () => {
  describe('templates', () => {
    test('is a Vue instance', () => {
      const wrapper = mount(SocialLinks);
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should create menu', () => {
      const wrapper = mount(SocialLinks);
      const links = wrapper.find('[data-social-links]');
      const facebook = wrapper.find('[data-social-links-facebook]');
      const facebookImg = wrapper.find('[data-social-links-facebook-img]');
      const twitter = wrapper.find('[data-social-links-twitter]');
      const twitterImg = wrapper.find('[data-social-links-twitter-img]');
      const instagram = wrapper.find('[data-social-links-instagram]');
      const instagramImg = wrapper.find('[data-social-links-instagram-img]');

      expect(links.exists()).toBeTruthy();
      expect(links.classes()).toContain('social');
      expect(facebook.exists()).toBeTruthy();
      expect(facebook.attributes().href).toEqual('#');
      expect(facebookImg.exists()).toBeTruthy();
      expect(facebookImg.attributes().src).toEqual(
        '~/assets/images/facebook.png'
      );
      expect(twitter.exists()).toBeTruthy();
      expect(twitter.attributes().href).toEqual('https://twitter.com/@videojs');
      expect(twitterImg.exists()).toBeTruthy();
      expect(twitterImg.attributes().src).toEqual(
        '~/assets/images/twitter.png'
      );
      expect(instagram.exists()).toBeTruthy();
      expect(instagram.attributes().href).toEqual('#');
      expect(instagramImg.exists()).toBeTruthy();
      expect(instagramImg.attributes().src).toEqual(
        '~/assets/images/instagram.png'
      );
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
