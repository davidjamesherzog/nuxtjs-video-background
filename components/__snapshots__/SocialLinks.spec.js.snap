// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks templates is a Vue instance 1`] = `
<ul
  class="social"
  data-social-links=""
>
  <li>
    <a
      data-social-links-facebook=""
      href="#"
    >
      <img
        data-social-links-facebook-img=""
        src="~/assets/images/facebook.png"
      />
    </a>
  </li>
   
  <li>
    <a
      data-social-links-twitter=""
      href="https://twitter.com/@videojs"
    >
      <img
        data-social-links-twitter-img=""
        src="~/assets/images/twitter.png"
      />
    </a>
  </li>
   
  <li>
    <a
      data-social-links-instagram=""
      href="#"
    >
      <img
        data-social-links-instagram-img=""
        src="~/assets/images/instagram.png"
      />
    </a>
  </li>
</ul>
`;

exports[`SocialLinks templates should create menu 1`] = `
<ul
  class="social"
  data-social-links=""
>
  <li>
    <a
      data-social-links-facebook=""
      href="#"
    >
      <img
        data-social-links-facebook-img=""
        src="~/assets/images/facebook.png"
      />
    </a>
  </li>
   
  <li>
    <a
      data-social-links-twitter=""
      href="https://twitter.com/@videojs"
    >
      <img
        data-social-links-twitter-img=""
        src="~/assets/images/twitter.png"
      />
    </a>
  </li>
   
  <li>
    <a
      data-social-links-instagram=""
      href="#"
    >
      <img
        data-social-links-instagram-img=""
        src="~/assets/images/instagram.png"
      />
    </a>
  </li>
</ul>
`;
